src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -538,21 +539,26 @@ public void acceptRecord(final SAMRecord record) {
                         boolean isOnTarget = false;
                         for (final Interval target : targets) {
                             if (refPos >= target.getStart() && refPos <= target.getEnd()) {
-                                isOnTarget = true;
                                 ++this.metrics.ON_TARGET_BASES;
                                 if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
 
                                 final int targetOffset = refPos - target.getStart();
                                 final Coverage coverage = this.coverageByTarget.get(target);
                                 coverage.addBase(targetOffset);
                                 baseQHistogramArray[baseQualities[offset]]++;
+                                coveredTargets.add(target);
+                                isOnTarget = true;
                             }
                         }
 
                         if (!isOnTarget) this.metrics.PCT_EXC_OFF_TARGET++;
                     }
                 }
             }
+            for (final Interval target : coveredTargets) {
```

Comment:
I've added a comment on line 549 (where the if-statement should be inserted). I think that might be bit clearer.